### DISCUSSION TITLE
Resource Action not working, returns 403 error

### DISCUSSION THREAD

**Discussion by Elshaden** [laravel/nova-issues#4989] [Q&A] [answered]
- Laravel Version: 9.37.#
- Nova Version: 4.17.1
- PHP Version: 8.1.#

### Description:
my resource action 

```
 public function actions(NovaRequest $request)
    {

            return [
            
                (new MyAction)->confirmText('Approval Required')
                    ->confirmButtonText('Approve')
                   ->canSee(function ($request) {
                        return  is_null($this->resource->approval);
                })
            ];
   }

```

The problem is when I click the Modal Opens but after submitting with approval_ref,  I get 403 not authorized.

My Policy 

```
 public function runAction(User $user, Order $order)
    {
        return true;
    }


```

 I am stuck without any solution to this problem which   I assume is a BUG.

**Comment by crynobone** [comment, marked as answer]
> I am stuck without any solution to this problem which I assume is a BUG.

We already have multiple feature and dusk tests covering this use case and everything working as expected. However I wouldn't recommend using `canSee` such as above and can imagine how that going to cause an issue since `$this->resource` not always yield a record. 

You should use `canRun(callable(\Laravel\Nova\Http\Requests\NovaRequest, \Illuminate\Database\Eloquent\Model))` instead.

**Comment by Elshaden** [comment]
Great Thanks, @crynobone 
This works, maybe worth documenting this scenario. 

appreciate your support.
